lib/dslash_coarse.hpp

```diff
@@ -214,214 +149,152 @@ namespace quda
-                         .instantiate(Type<Float>(), nDim, Ns, Nc, Mc, (int)tp.aux.x, (int)tp.aux.y, dslash, clover,
-                                      dagger, type, Type<Arg>())
-                         .configure(tp.grid, tp.block, tp.shared_bytes, stream)
-                         .launch(arg);
-#else  // !JITIFY
         switch (tp.aux.y) { // dimension gather parallelisation
         case 1:
           switch (tp.aux.x) { // this is color_col_stride
-          case 1:
-            qudaLaunchKernel(coarseDslashKernel<Float, nDim, Ns, Nc, Mc, 1, 1, dslash, clover, dagger, type, Arg>, tp,
-                             stream, arg);
-            break;
-          case 2:
-            qudaLaunchKernel(coarseDslashKernel<Float, nDim, Ns, Nc, Mc, 2, 1, dslash, clover, dagger, type, Arg>, tp,
-                             stream, arg);
-            break;
-          case 4:
-            qudaLaunchKernel(coarseDslashKernel<Float, nDim, Ns, Nc, Mc, 4, 1, dslash, clover, dagger, type, Arg>, tp,
-                             stream, arg);
-            break;
-          case 8:
-            qudaLaunchKernel(coarseDslashKernel<Float, nDim, Ns, Nc, Mc, 8, 1, dslash, clover, dagger, type, Arg>, tp,
-                             stream, arg);
-            break;
+          case 1: launch_device<CoarseDslash>(tp, stream, Arg<1, 1>(out, inA, inB, Y, X, (Float)kappa, parity)); break;
+#ifndef QUDA_FAST_COMPILE_DSLASH
+          case 2: launch_device<CoarseDslash>(tp, stream, Arg<2, 1>(out, inA, inB, Y, X, (Float)kappa, parity)); break;
+          case 4: launch_device<CoarseDslash>(tp, stream, Arg<4, 1>(out, inA, inB, Y, X, (Float)kappa, parity)); break;
+          case 8: launch_device<CoarseDslash>(tp, stream, Arg<8, 1>(out, inA, inB, Y, X, (Float)kappa, parity)); break;
+#endif
           default: errorQuda("Color column stride %d not valid", tp.aux.x);
           }
           break;
+#ifndef QUDA_FAST_COMPILE_DSLASH
         case 2:
           switch (tp.aux.x) { // this is color_col_stride
-          case 1:
-            qudaLaunchKernel(coarseDslashKernel<Float, nDim, Ns, Nc, Mc, 1, 2, dslash, clover, dagger, type, Arg>, tp,
-                             stream, arg);
-            break;
-          case 2:
-            qudaLaunchKernel(coarseDslashKernel<Float, nDim, Ns, Nc, Mc, 2, 2, dslash, clover, dagger, type, Arg>, tp,
-                             stream, arg);
-            break;
-          case 4:
-            qudaLaunchKernel(coarseDslashKernel<Float, nDim, Ns, Nc, Mc, 4, 2, dslash, clover, dagger, type, Arg>, tp,
-                             stream, arg);
-            break;
-          case 8:
-            qudaLaunchKernel(coarseDslashKernel<Float, nDim, Ns, Nc, Mc, 8, 2, dslash, clover, dagger, type, Arg>, tp,
-                             stream, arg);
-            break;
+          case 1: launch_device<CoarseDslash>(tp, stream, Arg<1, 2>(out, inA, inB, Y, X, (Float)kappa, parity)); break;
+#ifndef QUDA_FAST_COMPILE_DSLASH
```

Comment:
This `#ifndef` is redundant of the one above

Reply:
d45340b86